### ISSUE TITLE
chore: add .clang-format

### ISSUE THREAD

**Issue by carlocorradini** [karljj1/kdis#8]

**Comment by carlocorradini** [comment]
@karljj1 I've started a refactoring. Can you take a look at https://github.com/carlocorradini/kdis/tree/new 🥳

**Comment by karljj1** [comment]
Yeah sure. I'll take a look next week. Thanks for spending some time on this.

**Comment by carlocorradini** [comment]
@karljj1 Can I give you "edit" permissions? Soooo we can collaborate

**Comment by karljj1** [comment]
Yeah that's fine. I don't mind you working on a branch on the main repo if you like. Whatever you prefer.

**Comment by carlocorradini** [comment]
Done 🥳

**Comment by karljj1** [comment]
Hey,
Looks good so far. What is the reasons for using hpp instead of h for the file extensions?
You have my permission to fix the formatting ;) 
I long ago stopped liking the spaces in brackets I was using `if ( something )` or `void someMethod( float value )`.
So if you want to run some formatting tool to fix that I would be happy ;) 
I did use a tool to do it a long time ago but can't remember the name now.

**Comment by carlocorradini** [comment]
Working on it :)

For why .hpp is preferable see https://stackoverflow.com/a/152671/6676781

We don't provide C compatible headers so... keep them different
